### PR TITLE
added AR Advance for Create/Update

### PR DESCRIPTION
**Pull request by dylan-holt-sage**

### CODE REVIEW

File: Intacct.SDK.Tests/Functions/AccountsReceivable/ArAdvanceUpdateTest.cs [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+            <PAYMENTDATE>06/30/2015</PAYMENTDATE>
+            <RECEIPTDATE>06/30/2015</RECEIPTDATE>
+            <ARADVANCEITEMS>
+                <ARADVANCEITEM>
+                    <TRX_AMOUNT>76343.43</TRX_AMOUNT>
+                </ARADVANCEITEM>
+            </ARADVANCEITEMS>
+        </ARADVANCE>
+    </update>
+</function>";
```
Comment (pemba-sherpa-sage) [inline Intacct.SDK.Tests/Functions/AccountsReceivable/ArAdvanceUpdateTest.cs:50]: please fix the indentation here and in all other places for similar construct

Comment (dylan-holt-sage) [reply Intacct.SDK.Tests/Functions/AccountsReceivable/ArAdvanceUpdateTest.cs:50]: what's wrong with the indentation?

**Comment by pemba-sherpa-sage** [comment]
I will hold on to this one for now. Once you get your other PR with CI merged, please pull the latest on this branch and I review it together CI run results

File: Intacct.SDK.Tests/Functions/AccountsReceivable/ArAdvanceCreateTest.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
+﻿/*
+ * Copyright 2020 Sage Intacct, Inc.
```
Comment (pemba-sherpa-sage) [inline Intacct.SDK.Tests/Functions/AccountsReceivable/ArAdvanceCreateTest.cs:2]: see if you can update these to `2022` (here and in other new files as part of this PR) ; you don't have to update old ones

File: Intacct.SDK/Functions/AccountsReceivable/ArAdvanceCreate.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        }
+        
+        public override void WriteXml(ref IaXmlWriter xml)
+        {
+            xml.WriteStartElement("function");
+            xml.WriteAttribute("controlid", ControlId, true);
+
+            xml.WriteStartElement("create");
+            
+            xml.WriteStartElement("ARADVANCE");
```
Comment (pemba-sherpa-sage) [inline Intacct.SDK/Functions/AccountsReceivable/ArAdvanceCreate.cs:34]: see if these literals you are using like `ARADVANCE` , `CUSTOMERID`, etc.. already have a constant dictionary that you can use. If there are no constants defined for these, don't worry about it

Comment (dylan-holt-sage) [reply Intacct.SDK/Functions/AccountsReceivable/ArAdvanceCreate.cs:34]: There aren't any, unfortunately

**Review by pemba-sherpa-sage — APPROVED** [review]
+1 after CI is green and there is adequate coverage for new codes added

**Pull request merged**